modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/asset/KaleoProcessAssetRendererFactory.java

```diff
@@ -114,45 +114,23 @@ public boolean hasPermission(
 			permissionChecker, classPK, actionId);
 	}
 
-	@Reference(
-		target = "(osgi.web.symbolicname=com.liferay.portal.workflow.kaleo.forms.web)",
-		unbind = "-"
-	)
-	public void setServletContext(ServletContext servletContext) {
-		_servletContext = servletContext;
-	}
-
-	@Reference(unbind = "-")
-	protected void setDDLRecordLocalService(
-		DDLRecordLocalService ddlRecordLocalService) {
-
-		_ddlRecordLocalService = ddlRecordLocalService;
-	}
-
-	@Reference(unbind = "-")
-	protected void setKaleoProcessLinkLocalService(
-		KaleoProcessLinkLocalService kaleoProcessLinkLocalService) {
-
-		_kaleoProcessLinkLocalService = kaleoProcessLinkLocalService;
-	}
+	@Reference
+	private DDLRecordLocalService _ddlRecordLocalService;
 
-	@Reference(unbind = "-")
-	protected void setKaleoProcessLocalService(
-		KaleoProcessLocalService kaleoProcessLocalService) {
+	@Reference
```

Comment:
@W2jx This commit also just converts method references to field references. No "duplicate" to remove.